### PR TITLE
[SYCLomatic] Support migration of 30 half related Math APIs.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by yihwang-nv — APPROVED** [review]

File: clang/test/dpct/helper_files_ref/include/math.hpp [hunk 1]
```diff
@@ -149,6 +149,19 @@ compare(const T a, const T b, const BinaryOperation binary_op) {
   return {compare(a[0], b[0], binary_op), compare(a[1], b[1], binary_op)};
 }
 
+/// Performs 2 element comparison and return an unsigned int.
```
Comment (ShengchenJ) [inline clang/test/dpct/helper_files_ref/include/math.hpp]: performs 2 elements comparison and returns ..

Comment (tangjj11) [reply clang/test/dpct/helper_files_ref/include/math.hpp]: fixed

**Review by ziranzha — APPROVED** [review]

File: clang/test/dpct/helper_files_ref/include/math.hpp [hunk 2]
```diff
@@ -266,6 +293,70 @@ inline std::uint64_t max(const std::uint32_t a, const std::uint64_t b) {
...
+    ret[0] = 0.f;
+  if (!detail::isnan(ret[1]) && ret[1] < 0.f)
+    ret[1] = 0.f;
+  return ret;
+}
+
+/// Performs complex number multiply addition.
+/// \param [in] a The first value
+/// \param [in] b The second value
+/// \param [in] b The third value
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/math.hpp]:
```suggestion
/// \param [in] c The third value
```

File: clang/test/dpct/helper_files_ref/include/math.hpp [hunk 3]
```diff
@@ -149,6 +149,19 @@ compare(const T a, const T b, const BinaryOperation binary_op) {
   return {compare(a[0], b[0], binary_op), compare(a[1], b[1], binary_op)};
 }
 
+/// Performs 2 elements comparison and returns an unsigned int.
```
Comment (zhimingwang36) [inline clang/test/dpct/helper_files_ref/include/math.hpp]: Performs 2 elements comparison, compare result of each element is  0 (false) or 0xffff (true), returns an unsigned int by composing compare result of two elements.

File: clang/test/dpct/helper_files_ref/include/math.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/test/dpct/helper_files_ref/include/math.hpp]: fixed

File: clang/test/dpct/helper_files_ref/include/math.hpp (hunk 3 above)
Comment (tangjj11) [reply clang/test/dpct/helper_files_ref/include/math.hpp]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**